src/main/java/com/artipie/asto/Storage.java

```diff
@@ -50,7 +50,8 @@
 
     /**
      * Return the list of keys that start with this prefix, for
-     * example "foo/bar/".
+     * example "foo/bar/". Resulting list is sorted by string
```

Comment:
@olenagerasimova this method `list()` returns `Collection<>` interface which may don't support ordering, see https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html

> The root interface in the collection hierarchy. A collection represents a group of objects, known as its elements. Some collections allow duplicate elements and others do not. **Some are ordered and others unordered.** The JDK does not provide any direct implementations of this interface: it provides implementations of more specific subinterfaces like Set and List. This interface is typically used to pass collections around and manipulate them where maximum generality is desired.